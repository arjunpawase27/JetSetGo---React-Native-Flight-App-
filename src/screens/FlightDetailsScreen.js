import React, { useEffect, useRef } from 'react';
import { View, Text, StyleSheet, Animated } from 'react-native';

const FlightDetailsScreen = ({ route }) => {
  const { aircraft, airline, arrivalTime, departureTime, destination, duration, flightNumber, gate, origin, price, seatsAvailable } = route.params;

  const animations = useRef([
    new Animated.Value(0),
    new Animated.Value(0),
    new Animated.Value(0),
    new Animated.Value(0),
    new Animated.Value(0),
    new Animated.Value(0),
    new Animated.Value(0),
    new Animated.Value(0),
    new Animated.Value(0),
    new Animated.Value(0),
    new Animated.Value(0),
  ]).current;

  useEffect(() => {
    const animateDetails = () => {
      const animationSequence = animations.map((anim, index) => (
        Animated.timing(anim, {
          toValue: 1,
          duration: 500,
          useNativeDriver: true,
          delay: index * 100,
        })
      ));
      Animated.stagger(100, animationSequence).start();
    };

    animateDetails();
  }, [animations]);

  return (
    <View style={styles.container}>
      <View style={styles.innerContainer}>
        <View style={styles.innerBox}>
        <Text style={styles.title}>Flight Details</Text>

        {[
          { label: 'Aircraft:', value: aircraft },
          { label: 'Airline:', value: airline },
          { label: 'Departure Date:', value: departureTime },
          { label: 'Arrival Date:', value: arrivalTime },
          { label: 'Origin:', value: origin },
          { label: 'Destination:', value: destination },
          { label: 'Flight Number:', value: flightNumber },
          { label: 'Duration:', value: duration },
          { label: 'Gate:', value: gate },
          { label: 'Price:', value: `$${price}` },
          { label: 'Seats Available:', value: seatsAvailable },
        ].map((detail, index) => (
          <Animated.View key={index} style={[styles.detailContainer, { opacity: animations[index] }]}>
            <Text style={styles.detailLabel}>{detail.label}</Text>
            <Text style={styles.detailText}>{detail.value}</Text>
          </Animated.View>
        ))}
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#7bd3f7',
    
  },
  innerContainer: {
    flex: 1,
    padding: 20,
    paddingTop: 100,
    backgroundColor: '#f8f9fa',
    borderTopLeftRadius: 1000,
  },
  innerBox:{
    borderRadius:20,
    borderWidth:2,
    padding:12,
    borderColor:"#ced4da"
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 20,
    color: '#343a40',
  },
  detailContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
  },
  detailLabel: {
    fontSize: 16,
    fontWeight: 'bold',
    marginRight: 5,
    color: '#555',
  },
  detailText: {
    fontSize: 16,
    color: '#333',
  },
});

export default FlightDetailsScreen;
