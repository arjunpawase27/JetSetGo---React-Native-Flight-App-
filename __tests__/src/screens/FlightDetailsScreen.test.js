// __tests__/src/screens/FlightDetailsScreen.test.js

import React from 'react';
import { render } from '@testing-library/react-native';
import FlightDetailsScreen from '../../../src/screens/FlightDetailsScreen';
describe('FlightDetailsScreen', () => {
  test('renders correctly', () => {
    const route = {
      params: {
        aircraft: 'Boeing 787',
        airline: 'Air India',
        arrivalTime: '2024-03-15T20:30:00',
        departureTime: '2024-03-15T17:00:00',
        destination: 'Kolkata',
        duration: '3 hours 30 minutes',
        flightNumber: 'AI230',
        gate: 'D10',
        origin: 'Bangalore',
        price: 6200,
        seatsAvailable: 110,
      },
    };

    const { toJSON } = render(<FlightDetailsScreen route={route} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
