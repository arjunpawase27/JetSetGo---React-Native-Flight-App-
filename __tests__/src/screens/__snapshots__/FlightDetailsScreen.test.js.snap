// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightDetailsScreen renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#7bd3f7",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#f8f9fa",
        "borderTopLeftRadius": 1000,
        "flex": 1,
        "padding": 20,
        "paddingTop": 100,
      }
    }
  >
    <View
      style={
        {
          "borderColor": "#ced4da",
          "borderRadius": 20,
          "borderWidth": 2,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          {
            "color": "#343a40",
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 20,
          }
        }
      >
        Flight Details
      </Text>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
            "opacity": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 5,
            }
          }
        >
          Aircraft:
        </Text>
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
            }
          }
        >
          Boeing 787
        </Text>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
            "opacity": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 5,
            }
          }
        >
          Airline:
        </Text>
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
            }
          }
        >
          Air India
        </Text>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
            "opacity": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 5,
            }
          }
        >
          Departure Date:
        </Text>
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
            }
          }
        >
          2024-03-15T17:00:00
        </Text>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
            "opacity": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 5,
            }
          }
        >
          Arrival Date:
        </Text>
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
            }
          }
        >
          2024-03-15T20:30:00
        </Text>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
            "opacity": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 5,
            }
          }
        >
          Origin:
        </Text>
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
            }
          }
        >
          Bangalore
        </Text>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
            "opacity": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 5,
            }
          }
        >
          Destination:
        </Text>
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
            }
          }
        >
          Kolkata
        </Text>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
            "opacity": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 5,
            }
          }
        >
          Flight Number:
        </Text>
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
            }
          }
        >
          AI230
        </Text>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
            "opacity": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 5,
            }
          }
        >
          Duration:
        </Text>
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
            }
          }
        >
          3 hours 30 minutes
        </Text>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
            "opacity": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 5,
            }
          }
        >
          Gate:
        </Text>
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
            }
          }
        >
          D10
        </Text>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
            "opacity": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 5,
            }
          }
        >
          Price:
        </Text>
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
            }
          }
        >
          $6200
        </Text>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
            "opacity": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 5,
            }
          }
        >
          Seats Available:
        </Text>
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
            }
          }
        >
          110
        </Text>
      </View>
    </View>
  </View>
</View>
`;
